packages/client/test/rpc/eth/getStorageAt.spec.ts

```diff
@@ -1,154 +1,53 @@
-    new Account()
-  )
-  t.notEqual(execution, undefined, 'should have valid execution')
-  const { vm } = execution
+  const address = Address.fromString(`0x${'11'.repeat(20)}`)
+  const emptySlotStr = `0x${'00'.repeat(32)}`
 
-  // genesis address with balance
-  const address = Address.fromString('0xccfd725760a68823ff1e062f4cc97e1360e8d997')
+  const { execution, server } = await setupChain(pow, 'pow')
 
-  // contract inspired from https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getstorageat/
-  /*
-    // SPDX-License-Identifier: MIT
-    pragma solidity ^0.7.4;
-
-    contract Storage {
-        uint pos0;
-        mapping(address => uint) pos1;
-
-        function store() public {
-            pos0 = 1234;
-            pos1[msg.sender] = 5678;
-        }
-    }
-  */
-  const data =
-    '0x6080604052348015600f57600080fd5b5060bc8061001e6000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c8063975057e714602d575b600080fd5b60336035565b005b6104d260008190555061162e600160003373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019081526020016000208190555056fea2646970667358221220b16fe0abdbdcae31fa05c5717ebc442024b20fb637907d1a05547ea2d8ec8e5964736f6c63430007060033'
-
-  // construct block with tx
-  const gasLimit = 2000000
-  const tx = Transaction.fromTxData({ gasLimit, data }, { common, freeze: false })
-  tx.getSenderAddress = () => {
-    return address
+  let req = params(method, [address.toString(), '0x0', 'latest'])
+  let expectRes = (res: any) => {
+    const msg = 'should return the empty slot for nonexistent account'
+    t.equal(res.body.result, emptySlotStr, msg)
   }
-  const parent = await blockchain.getCanonicalHeadHeader()
-  const block = Block.fromBlockData(
-    {
-      header: {
-        parentHash: parent.hash(),
-        number: 1,
-        gasLimit,
-      },
-    },
-    { common, calcDifficultyFromHeader: parent }
-  )
-  block.transactions[0] = tx
+  await baseRequest(t, server, req, 200, expectRes, false)
 
-  // deploy contract
-  let ranBlock: Block | undefined = undefined
-  vm.events.once('afterBlock', (result: any) => (ranBlock = result.block))
-  const result = await vm.runBlock({ block, generate: true, skipBlockValidation: true })
-  const { createdAddress } = result.results[0]
-  await vm.blockchain.putBlock(ranBlock!)
+  await execution.vm.stateManager.putAccount(address, new Account())
 
-  // call store() method
-  const funcHash = '975057e7' // store()
-  const storeTxData = {
-    to: createdAddress!.toString(),
-    from: address.toString(),
-    data: `0x${funcHash}`,
-    gasLimit: bigIntToHex(BigInt(530000)),
-    nonce: 1,
-  }
-  const storeTx = Transaction.fromTxData(storeTxData, { common, freeze: false })
-  storeTx.getSenderAddress = () => {
-    return address
+  req = params(method, [address.toString(), '0x0', 'latest'])
+  expectRes = (res: any) => {
```

Comment:
currently this is returning empty slot value because of 'nonexistent account' as well.

Reply:
I think "putAccount" assumes that it will be not be a contract.  

replace: `await execution.vm.stateManager.putAccount(...)` 
with:  `await execution.vm.stateManager.putContractCode(address, Uint8Array.from([]))
`

will fix this test and the next.
but should we change the way `putAccount` works instead?

Reply:
hmm, i thought it was an issue with the test, but i cant seem to fix it